src/main/scala/com/softwaremill/realworld/articles/core/ArticlesService.scala

```diff
@@ -52,10 +49,7 @@ class ArticlesService(
   def create(createData: ArticleCreateData, email: String): Task[Article] =
     for {
       userId <- userIdByEmail(email)
-      articleId <- articlesRepository.add(createData, userId)
-      _ <- ZIO.foreach(createData.tagList) { tagList =>
-        ZIO.foreach(tagList)(tag => articlesRepository.addTag(tag, articleId))
-      }
+      _ <- articlesRepository.addArticleTransaction(createData, userId)
```

Comment:
**suggestion:** Let's keep the `addArticle` name, no need to add "transaction" to the method name.